### ISSUE TITLE
HeatmapWidgets.py: "OverflowError: cannot convert float infinity to integer"

### ISSUE THREAD

**Issue by Geremia** [qgis/QGIS#56408]
### What is the bug or the crash?

```
File "/usr/share/qgis/python/plugins/processing/algs/qgis/ui/HeatmapWidgets.py", line 105, in pixelSizeChanged
              rows = max(round(self.raster_bounds.height() / cell_size) + 1, 1)
             OverflowError: cannot convert float infinity to integer
```

### Steps to reproduce the issue

Go to "[Heatmap (kernel density estimation)](https://docs.qgis.org/3.34/en/docs/user_manual/processing_algs/qgis/interpolation.html#qgisheatmapkerneldensityestimation)"

### Versions

Versione di QGIS
3.34.0-Prizren
branch codice QGIS
Release 3.34
Compilato con Qt
5.15.11
Esecuzione con Qt
5.15.12
Versione Python
3.9.18
Versione GDAL/OGR
3.7.3
Versione PROJ
9.3.1
Versione database del Registro EPSG
v10.098 (2023-11-24)
Versione GEOS
3.12.1-CAPI-1.18.1
Compilato su SQLite
3.44.2
In esecuzione su SQLite
3.45.1
Versione client PostgreSQL
unknown
Versione SpatiaLite
5.1.0
Versione QWT
6.1.6
Versione QScintilla2
2.11.6
Versione SO
Slackware 15.0 x86_64 (post 15.0 -current)

Plugins Python attivi
MetaSearch
0.3.6
db_manager
0.1.20
grassprovider
2.12.99
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by nyalldawson** [comment]
This was fixed in 3.34.1, please update your qgis.

**Comment by Geremia** [comment]
@nyalldawson I had a pip3-installed version of `sip` that was conflicting with my system version, which is why I couldn't compile 3.34.3, but now it builds. 😄
